src/containerapp/azext_containerapp/containerapp_env_decorator.py

```diff
@@ -132,6 +132,19 @@ def set_up_workload_profiles(self):
                     "maximumCount": 1
                 }
                 workload_profiles.append(gpu_profile)
+            if self.is_env_for_azml_app() and not self.get_argument_enable_dedicated_gpu():
+                if self.get_argument_workload_profile_type() is None or self.get_argument_workload_profile_type().lower() == "consumption-gpu-nc24-a100":
```

Comment:
>elf.get_argument_workload_profile_type() [](http://example.com/codeflow?start=20&length=40)

we're calling this multiple times. store in a variable.
wp_type = self.get_argument_workload_profile_type()